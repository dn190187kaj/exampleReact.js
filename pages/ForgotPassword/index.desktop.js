import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import { TextField, RaisedButton } from 'material-ui';
import injectSheet from 'react-jss';

import { LOG_IN, SIGN_UP } from 'consts/routes';

import styles from './styles/desktop';

@injectSheet(styles.jss)
export default class ForgotPassword extends PureComponent {
  static propTypes = {
    emailError: PropTypes.string,
    classes: PropTypes.shape(), // eslint-disable-line react/require-default-props
    forgotPassword: PropTypes.func,
  };

  static defaultProps = {
    emailError: '',
    forgotPassword: () => {},
  };

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      stage: 1,
    };
  }

  submitEmail = (event) => {
    event.preventDefault();
    this.props.forgotPassword(this.state.email).then((status) => {
      if (status) {
        this.setState({ stage: 2 });
      }
    });
  };

  render() {
    const { classes } = this.props;
    const { email, stage } = this.state;
    return (
      <div>
        <div className={classes.forgotPasswordMain}>
          <h1>Forgot Password</h1>
          {stage === 1 && (
            <div>
              <form className={classes.formField}>
                <TextField
                  id="login-input"
                  floatingLabelText="Your Email"
                  onChange={e => this.setState({ email: e.target.value })}
                  errorText={this.props.emailError}
                  style={styles.textField}
                  underlineStyle={styles.underlineStyle}
                /><br />
                <RaisedButton
                  className={classes.loginBtn}
                  label="Enter"
                  primary
                  type="submit"
                  onClick={this.submitEmail}
                />
                <div className={classes.loginLink}>
                  <span>
                    <Link to={LOG_IN}>
                      Login
                    </Link>
                  </span>
                  <span>
                    <Link to={SIGN_UP}>
                      Register
                    </Link>
                  </span>
                </div>
              </form>
            </div>
          )}
          {stage === 2 && (
            <div className={classes.formField}>
              <h2>
                We&apos;ve just sent you an email with instructions
                to the address - {email}, please check your email.
              </h2>
            </div>
          )}
        </div>
      </div>
    );
  }
}
