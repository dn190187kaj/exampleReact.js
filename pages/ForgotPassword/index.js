import React from 'react';
import { connect } from 'react-redux';
import { forgotPassword } from 'schemes/user/actions';
import { getEmailError } from 'schemes/user/selectors';
import { IS_MOBILE } from 'consts/browser';
import Desktop from './index.desktop';
import Mobile from './index.mobile';

export const ForgotPasswordContainer = (props) => {
  const ForgotPassword = IS_MOBILE ? Mobile : Desktop;
  return (
    <ForgotPassword {...props} />
  );
};

export default connect(
  state => ({
    emailError: getEmailError(state),
  }),
  dispatch => ({
    forgotPassword: email => dispatch(forgotPassword(email)),
  }),
)(ForgotPasswordContainer);
