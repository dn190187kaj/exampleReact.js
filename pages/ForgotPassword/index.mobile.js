import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';
import { Link } from 'react-router';
import injectSheet from 'react-jss';
import classNames from 'classnames';

import { LOG_IN, SIGN_UP } from 'consts/routes';

import styles from './styles/mobile';

@injectSheet(styles.jss)
export default class ForgotPassword extends PureComponent {
  static propTypes = {
    emailError: PropTypes.string,
    classes: PropTypes.shape(), // eslint-disable-line react/require-default-props
    forgotPassword: PropTypes.func,
  };

  static defaultProps = {
    emailError: '',
    forgotPassword: () => {},
  };

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      stage: 1,
    };
  }

  submitEmail(event) {
    event.preventDefault();
    return this.props.forgotPassword(this.state.email)
      .then((status) => {
        if (status) {
          this.setState({ stage: 2 });
        } else {
          this.setState({ stage: 1 });
        }
      });
  }

  render() {
    const { classes } = this.props;
    const { email, stage } = this.state;
    return (
      <div className={classes.forgotPasswordPage}>
        <div className={classes.forgotPasswordMainMob}>
          <h1>Forgot Password</h1>
          {stage === 1 && (
            <div>
              <form className={classes.formField}>
                <TextField
                  id="login-input"
                  hintText="Your Email"
                  hintStyle={styles.textFieldHint}
                  onChange={e => this.setState({ email: e.target.value })}
                  errorText={this.props.emailError}
                  errorStyle={styles.textFieldError}
                  style={styles.textField}
                  inputStyle={styles.textFieldInput}
                  underlineStyle={styles.underlineStyle}
                /><br />
                <RaisedButton
                  className={classes.loginBtn}
                  label="Enter"
                  primary
                  type="submit"
                  onClick={e => this.submitEmail(e)}
                  style={styles.loginBtnStyle}
                />
                <div className={classes.loginLink}>
                  <span>
                    <Link to={LOG_IN}>
                      Login
                    </Link>
                  </span>
                  <span>
                    <Link to={SIGN_UP}>
                      Register
                    </Link>
                  </span>
                </div>
              </form>
            </div>
          )}
          {stage === 2 && (
            <div className={classNames(classes.stage2, classes.formField)}>
              <h2>
                We&apos;ve just sent you an email with instructions
                to the address - {email}, please check your email.
              </h2>
            </div>
          )}
        </div>
      </div>
    );
  }
}
