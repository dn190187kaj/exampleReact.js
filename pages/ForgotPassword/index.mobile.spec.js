import React from 'react';
import { shallow } from 'enzyme';
import ForgotPassword from './index.mobile';

describe('<ForgotPassword /> mobile version', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      emailError: '',
      forgotPassword: () => Promise.resolve('ok'),
    };
    wrapper = shallow(<ForgotPassword {...props} />);
  });

  it('should render self and subcomponents', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show errors from props', () => {
    wrapper.setProps({
      emailError: 'email error!',
    });
    expect(wrapper.find('#login-input').props().errorText).toEqual('email error!');
  });
});
