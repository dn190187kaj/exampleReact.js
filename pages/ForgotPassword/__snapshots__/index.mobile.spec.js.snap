exports[`<ForgotPassword /> mobile version should render self and subcomponents 1`] = `
<div
  className="forgot-password-page">
  <Header
    type="registration" />
  <div
    className="forgot-password-main-mob">
    <h1>
      Forgot Password
    </h1>
    <div
      className="stage-1">
      <form
        className="form-field">
        <TextField
          className="login-input"
          disabled={false}
          errorStyle={
            Object {
              "fontSize": "12px",
              "marginTop": "-24px",
              "top": "24px",
            }
          }
          errorText=""
          floatingLabelFixed={false}
          fullWidth={false}
          hintStyle={
            Object {
              "bottom": "12px",
              "left": "0",
              "right": "0",
            }
          }
          hintText="Your Email"
          inputStyle={
            Object {
              "textAlign": "center",
            }
          }
          multiLine={false}
          onChange={[Function]}
          rows={1}
          style={
            Object {
              "fontSize": "16px",
              "height": "48px",
              "lineHeight": "24px",
              "width": "95vw",
            }
          }
          type="text"
          underlineShow={true}
          underlineStyle={
            Object {
              "borderColor": "#8d8d8d",
            }
          } />
        <br />
        <RaisedButton
          className="login-btn"
          disabled={false}
          fullWidth={false}
          label="Enter"
          labelPosition="after"
          onClick={[Function]}
          primary={true}
          secondary={false}
          style={
            Object {
              "fontSize": "16px",
              "height": "36px",
              "lineHeight": "36px",
              "marginTop": "60px",
              "width": "45vw",
            }
          }
          type="submit" />
        <div
          className="login-link">
          <span>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/login">
              Login
            </Link>
          </span>
          <span>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/registration">
              Register
            </Link>
          </span>
        </div>
      </form>
    </div>
  </div>
  <LoginAndSignUpFooterContainer />
</div>
`;
